### DISCUSSION TITLE
Data not saving to datasaver

### DISCUSSION THREAD

**Discussion by Diddleydope** [microsoft/Qcodes#5980] [Q&A]
Hi there! I'm having a problem with testing my QCoDes driver. In the following code, I am trying to plot some data:

```
testInstrument = NanonisTramea('TestInstance', '10.0.0.37', 6501) #QCODES DRIVER INSTANCE


initialise_or_create_database_at(Path.cwd() / "offline_plotting_example.db")
exp = load_or_create_experiment('offline_plotting_experiment', 'nosample')
meas = Measurement(exp=exp)
meas.register_parameter(testInstrument.Output1)
meas.register_parameter(testInstrument.Input0, setpoints=(testInstrument.Output1,))

with meas.run() as datasaver:
    for set_v in np.linspace(0, 10, 10):
        testInstrument.Output1.set(set_v)
        get_v = testInstrument.Input0.get()
        datasaver.add_result((testInstrument.Output1, set_v),(testInstrument.Input0, get_v))

    ds = datasaver.dataset
    ax, cbax = plot_dataset(ds)
```

Weirdly enough, the code executes without errors, but nothing happens. Turns out, the datasaver is empty. When i print out its contents like this:

```
print(ds.get_parameter_data())
```

I get this:

{'TestInstance_Input0': {'TestInstance_Input0': array([], dtype=float64), 'TestInstance_Output1': array([], dtype=float64)}}

So two empty arrays.
Can anyone explain why this is happening? I greatly appreciate any help!

**Comment by astafan8** [comment]
@Diddleydope hi! Looking at the snippet I can't see what the problem is. Just in case, when you call `testInstrument.Input0.get()` do you get an empty array or array with data?

**Reply by Diddleydope** [reply]
I get a singular value. For example, if I do `print(get_v)` after assigning it, I get a number like 3.0220062399166636e-06. Does that answer your question?
